src/libsync/ocsprofileconnector.cpp

```diff
@@ -0,0 +1,148 @@
+#include "ocsprofileconnector.h"
+#include "common/result.h"
+#include "networkjobs.h"
+#include "iconjob.h"
+#include "theme.h"
+
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QJsonArray>
+#include <QLoggingCategory>
+#include <QIcon>
+#include <QPainter>
+#include <QImage>
+#include <QSvgRenderer>
+#include <qpixmap.h>
+
+namespace {
+Q_LOGGING_CATEGORY(lcOcsProfileConnector, "nextcloud.gui.ocsprofileconnector", QtInfoMsg)
+
+OCC::HovercardAction jsonToAction(const QJsonObject &jsonActionObject)
+{
+    return {jsonActionObject.value("title").toString("No title"), jsonActionObject.value("icon").toString("no-icon"),
+        jsonActionObject.value("hyperlink").toString("no-link")};
+}
+
+OCC::Hovercard jsonToHovercard(const QJsonArray &jsonDataArray)
+{
+    OCC::Hovercard hovercard;
+    hovercard._actions.reserve(jsonDataArray.size());
+    for (const auto &jsonEntry : jsonDataArray) {
+        Q_ASSERT(jsonEntry.isObject());
+        if (!jsonEntry.isObject()) {
+            continue;
+        }
+        hovercard._actions.push_back(jsonToAction(jsonEntry.toObject()));
+    }
+    return hovercard;
+}
+
+OCC::Optional<QPixmap> createPixmapFromSvgData(const QByteArray &iconData)
+{
+    QSvgRenderer svgRenderer;
+    if (svgRenderer.load(iconData)) {
+        QSize imageSize{16, 16};
+        if (OCC::Theme::isHidpi()) {
+            imageSize = QSize{32, 32};
+        }
+        QImage scaledSvg(imageSize, QImage::Format_ARGB32);
+        scaledSvg.fill("transparent");
+        QPainter svgPainter{&scaledSvg};
+        svgRenderer.render(&svgPainter);
+        return QPixmap::fromImage(scaledSvg);
+    }
+    return {};
+}
+
+OCC::Optional<QPixmap> iconDataToPixmap(const QByteArray iconData)
+{
+    if (!iconData.startsWith("<svg")) {
+        return {};
+    }
+    return createPixmapFromSvgData(iconData);
+}
+}
+
+namespace OCC {
+
+HovercardAction::HovercardAction() = default;
+
+HovercardAction::HovercardAction(QString title, QUrl iconUrl, QUrl link)
+    : _title(std::move(title))
+    , _iconUrl(std::move(iconUrl))
+    , _link(std::move(link))
+{
+}
+
+OcsProfileConnector::OcsProfileConnector(AccountPtr account, QObject *parent)
+    : QObject(parent)
+    , _account(account)
+{
+}
+
+void OcsProfileConnector::fetchHovercard(const QString &userId)
+{
+    const QString url = QStringLiteral("/ocs/v2.php/hovercard/v1/%1").arg(userId);
```

Comment:
are you sure every versions of the nextcloud server are including support for this API ?
you should add a capability check of request such a thing from server
currently I just get error messages in the log

Reply:
Good point, I'm not sure there is a capability for this. I need to check.

Reply:
This endpoint will always be available with NC23. I will check for this version.

Reply:
@FlexW `const auto url `

Reply:
> This endpoint will always be available with NC23. I will check for this version. There is no capability for it.

is'nt something we should forward to Tobias and ask for such a capability ?

Reply:
I originally asked Tobias how to handle it. He said just check for version 23. As far as I understand capabilities are for stuff that may be optionally available. It seems like that this profile page is built into the server, so the route is always there.